### PR TITLE
Add archlinux jdk11 Dockerfile

### PR DESCRIPTION
**Pull request by soloturn**
add a dockerfile for arch linux to have a rolling release image with newest versions. can be made with openjdk14 or 15 if desired:
* https://www.archlinux.org/packages/extra/x86_64/jdk-openjdk/
* https://aur.archlinux.org/packages/java-openjdk-ea-bin/

### CODE REVIEW

File: 11/arch/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ARG uid=1000
+ARG gid=1000
+
+RUN groupadd -g ${gid} ${group}
+RUN useradd -c "Jenkins user" -d /home/${user} -u ${uid} -g ${gid} -m ${user}
+LABEL Description="This is a base image, which provides the Jenkins agent executable (agent.jar)" Vendor="Jenkins project" Version="${VERSION}"
+
+ARG AGENT_WORKDIR=/home/${user}/agent
+
+RUN pacman -Syu git jdk11-openjdk --noconfirm
```
Comment (slide) [inline 11/arch/Dockerfile:37]: Does installing git also install git lfs? We generally install that on agent images.

Comment (fossdd) [reply 11/arch/Dockerfile:37]: No, it doesn't.
```suggestion
RUN pacman -Syu git git-lfs jdk11-openjdk --noconfirm
```

**Comment by fossdd** [comment]
I suggest calling the directories, name,.. `archlinux` instead of `arch`.

`arch` also could mean the architecture and it is more colloquial.

**Comment by slide** [comment]
Are you willing to maintain the image? We are not planning on adding new images unless there is a maintainer for them.

**Comment by fossdd** [comment]
Yes, I think I could maintain it.

**Comment by slide** [comment]
Looks like there are conflicts that need to be resolved.

**Comment by fossdd** [comment]
Yes, then this here should be used for the makefile.
```patch
----------------------------------- Makefile -----------------------------------
index 564553f..bfb4369 100644
@@ -4,9 +4,9 @@ IMAGE_NAME:=jenkins4eval/agent
 IMAGE_NAME_AGENT:=jenkins4eval/slave
 
 .PHONY: build
-.PHONY: test test-alpine test-debian test-debian-buster test-jdk11 test-jdk11-alpine test-jdk11-buster
+.PHONY: test test-alpine test-arch test-debian test-debian-buster test-jdk11 test-jdk11-alpine test-jdk11-buster
 
-build: build-alpine build-debian build-debian-buster build-jdk11 build-jdk11-alpine build-jdk11-buster
+build: build-alpine build-arch build-debian build-debian-buster build-jdk11 build-jdk11-alpine build-jdk11-buster
 
 build-alpine:
 	docker build -t ${IMAGE_NAME}:alpine \
@@ -15,6 +15,12 @@ build-alpine:
                  -t ${IMAGE_NAME_AGENT}:alpine \
                  8/alpine/
 
+build-arch:
+	docker build -t ${IMAGE_NAME}:latest \
+                 -t ${IMAGE_NAME}:arch \
+                 -t ${IMAGE_NAME}:jdk11-arch \
+                 11/arch/
+
 build-debian:
 	docker build -t ${IMAGE_NAME}:stretch \
                  -t ${IMAGE_NAME}:jdk8-stretch \
@@ -59,7 +65,7 @@ test-run-%: build prepare-test
 					-w /usr/src/app node:12-alpine \
 					sh -c "npm install tap-xunit -g && cat target/results-$*.tap | tap-xunit --package='jenkinsci.docker-agent.$*' > target/junit-results-$*.xml"
 
-test: test-alpine test-debian test-debian-buster test-jdk11 test-jdk11-buster
+test: test-alpine test-arch test-debian test-debian-buster test-jdk11 test-jdk11-buster
 
 test-alpine: FOLDER=8/alpine
 test-alpine: test-run-alpine
@@ -67,6 +73,9 @@ test-alpine: test-run-alpine
 test-debian: FOLDER=8/stretch
 test-debian: test-run-debian
 
+test-arch: FOLDER=11/arch
+test-arch: test-run-arch
+
 test-debian-buster: FOLDER=8/buster
 test-debian-buster: test-run-debian-buster
 ```

**Comment by slide** [comment]
Superseded by #171?

**Comment by fossdd** [comment]
I think yes

**Pull request closed without merging**